File: 671-second-minimum-node-in-a-binary-tree/671-second-minimum-node-in-a-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def findSecondMinimumValue(self, root: Optional[TreeNode]) -> int:
        u=set()
        stack=[root]
        while stack:
            x=stack.pop()
            if x:
                u.add(x.val)
                stack.append( x.left )
                stack.append( x.right )
        if len(u)<2:
            return -1
        return sorted(u)[1]

File: 671-second-minimum-node-in-a-binary-tree/671-second-minimum-node-in-a-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int>v;
    void com(TreeNode* root){
        if(root==NULL)return;
        v.push_back(root->val);
        com(root->left);
        com(root->right);
    }
    int findSecondMinimumValue(TreeNode* root) {
        com(root);
        sort(begin(v),end(v));
        for(int i=1;i<v.size();++i){
            if(v[i]!=v[0]){
                return v[i];
            }
        }
        return -1;
    }
};